dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIEJBManagedObjectFactoryImpl.java

```diff
@@ -129,53 +189,24 @@ public ManagedObjectContext createContext() {
 
     private synchronized EjbDescriptor<T> getEjbDescriptor() {
         if (this.ejbDescriptor == null) {
-
-            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                Tr.debug(tc, "Looking for EJB Bean: " + this.ejbName);
-            }
-            WebSphereBeanDeploymentArchive bda = super.getCurrentBeanDeploymentArchive();
-            WeldManager beanManager = null;
-            if (bda != null) {
-                beanManager = (WeldManager) bda.getBeanManager();
-            } else {
-                beanManager = (WeldManager) getCDIRuntime().getCurrentBeanManager();
-            }
-
-            EjbDescriptor<T> ejbDescriptor = beanManager.getEjbDescriptor(this.ejbName);
-
-            if (ejbDescriptor == null) {
-                Set<WebSphereBeanDeploymentArchive> children = bda.getDescendantBdas();
-                Iterator<WebSphereBeanDeploymentArchive> itr = children.iterator();
-                while (ejbDescriptor == null && itr.hasNext()) {
-                    bda = itr.next();
-                    beanManager = (WeldManager) bda.getBeanManager();
-                    ejbDescriptor = beanManager.getEjbDescriptor(this.ejbName);
-                }
-                // check all accessible BDAs if we haven't found the ejbDescriptor in descendant BDAs
-                if (ejbDescriptor == null) {
-                    children = bda.getWebSphereBeanDeploymentArchives();
-                    itr = children.iterator();
-                    while (ejbDescriptor == null && itr.hasNext()) {
-                        bda = itr.next();
-                        beanManager = (WeldManager) bda.getBeanManager();
-                        ejbDescriptor = beanManager.getEjbDescriptor(this.ejbName);
-                    }
-                }
-            }
-
-            if (ejbDescriptor != null) {
-                this.ejbDescriptor = ejbDescriptor;
-                this.ejbBeanManager = beanManager;
-                this.ejbBDA = bda;
-            }
-
+            getBean();
         }
+        return this.ejbDescriptor;
+    }
 
-        if (this.ejbDescriptor == null) {
-            Tr.error(tc, "Could not find an EjbDescriptor for : " + this.ejbName);
+    private boolean getBeanFromDescriptor(EjbDescriptor<T> descriptor, WeldManager beanManager)
+    {
+        if (ejbDescriptor != null) {
+            Bean bean = (Bean<T>) beanManager.getBean(ejbDescriptor);
```

Comment:
There's no need for the cast here, and you can use `Bean<T>` instead of `Bean`.
```suggestion
            Bean<T> bean = beanManager.getBean(ejbDescriptor);
```